activerecord/test/models/admin/region.rb

```diff
@@ -0,0 +1,6 @@
+# The class Region (without namespace) shouldn't exist
+# in order to the test test_type_mismatch_with_namespaced_class
+# be valid
+class Admin::Region < ActiveRecord::Base
```

Comment:
The test should be done with a class that only exists in a namespace, but maybe a possibility is to undefine `Account` and use `Admin::Account`. Do you think is better?

Also, I added a new validation to the test to ensure that it is really testing what it should.